packages/donor/src/components/AppHeader/AppHeader.tsx

```diff
@@ -45,6 +53,13 @@ export default function ButtonAppBar({
     <AppBar position="fixed" color="secondary">
       <Toolbar>
         <Grid container>
+          <Grid item xs>
+            {hasBurgerMenu && (
+              <BurgerMenu
+                onClick={() => setShowSideBar((previous) => !previous)}
```

Comment:
isnt this enough?
`onClick={() => setShowSideBar(!showSideBar)}`

Reply:
The correct way to update a state based on the previous state in react is to use the overloaded that gives you the previous state and not rely on the state value.